fuse/core.go

```diff
@@ -244,8 +244,14 @@ func ServeFuseFS(
 
 			log.Infof("FUSE: Unmounting %v", mountpoint)
 			if err = fuse.Unmount(mountpoint); err != nil {
-				log.Warnf("FUSE: Shutdown failed: %v", err.Error())
+				log.Warnf("FUSE: Shutdown failed: %v", err)
 				log.Warnf("FUSE: Manual cleanup required: umount %v", mountpoint)
+
+				// Retry in a loop until no longer blocked buy an open handle.
+				// All errors are `*os.PathError`, so we just match a known error string.
+				for ; err != nil && err.Error() == "resource busy"; err = fuse.Unmount(mountpoint) {
```

Comment:
Is "resource busy" platform agnostic? Also, what are the chances that this could be an infinite loop (i.e. should there be a counter)?

Reply:
I'm not sure about platform agnostic, I'll try it out on a linux machine.

I'm pretty sure it's not an infinite loop. Once the fs is unmounted, the error message changes to `invalid argument`.

Reply:
Ah, it's `Resource busy` on Linux :)

Reply:
Updated so it now works on Linux and macOS.

Reply:
Re: infinite loop, I'm mostly worried about those open handles never closing so resource busy is always returned. Sounds like that's not an issue?

Reply:
Well, the current behavior is it doesn't exit in that case either.

Reply:
Maybe a warning could be useful? Least let the user know that something went wrong with the unmount?

Reply:
That's what happens 2 lines up.